Enforce enum : bool { No, Yes } order

squelart opened the pull request:
#### f2554a7bf3fafaf732e9ea2f5584ffa1a8a6616f
<pre>
Enforce enum : bool { No, Yes } order
<a href="https://bugs.webkit.org/show_bug.cgi?id=282674">https://bugs.webkit.org/show_bug.cgi?id=282674</a>
<a href="https://rdar.apple.com/problem/139335415">rdar://problem/139335415</a>

Reviewed by Anne van Kesteren.

This ensures that casting to/from the underlying bool matches
the expected values No==0==false and Yes==1==true.

check-webkit-style now catches unexpected {Yes,No} ordering.

A few instances were caught and fixed.

* Source/WTF/wtf/text/Base64.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Background.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperAnimation.h:
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
* Source/WebKit/Shared/ColorControlSupportsAlpha.h:
* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(_EnumState.__init__):
(_EnumState.process_clean_line):
(_EnumState.process_clean_line.check_case_error):
(_EnumState.process_clean_line.check_case_error.is_case_error):
(_EnumState.process_clean_line.check_member_list):
(_EnumState):
(check_enum_members):
(check_style):
(CppChecker):
(_EnumState.process_clean_line.is_case_error): Deleted.
(check_enum_casing): Deleted.
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(NoNonVirtualDestructorsTest):

Canonical link: <a href="https://commits.webkit.org/286492@main">https://commits.webkit.org/286492@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/35f49e7776c8cedd88148466105f34454f1d418d

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75920 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/54949 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28817 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80417 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27186 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/78036 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64091 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3243 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59527 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17686 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78987 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49407 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65202 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39887 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/75427 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/46807 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22686 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25513 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/69097 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/67922 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23023 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81881 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/75196 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3289 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/2082 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67756 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3443 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65170 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67065 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/16758 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11011 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9136 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/97450 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3239 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/6045 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21310 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3260 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/4198 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/3267 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
I was wondering about this well after I landed ColorControlSupportsAlpha.

This seems like a reasonable change and makes us more robust and is probably minor enough that it doesn't need preflighting on webkit-dev or be explicitly added to the style guide.

squelart committed:
Enforce enum : bool { No, Yes } order
https://bugs.webkit.org/show_bug.cgi?id=282674
rdar://problem/139335415

Reviewed by Anne van Kesteren.

This ensures that casting to/from the underlying bool matches
the expected values No==0==false and Yes==1==true.

check-webkit-style now catches unexpected {Yes,No} ordering.

A few instances were caught and fixed.

* Source/WTF/wtf/text/Base64.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Background.cpp:
* Source/WebCore/dom/Document.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperAnimation.h:
* Source/WebCore/rendering/updating/RenderTreeUpdater.cpp:
* Source/WebKit/Shared/ColorControlSupportsAlpha.h:
* Tools/Scripts/webkitpy/style/checkers/cpp.py:
(_EnumState.__init__):
(_EnumState.process_clean_line):
(_EnumState.process_clean_line.check_case_error):
(_EnumState.process_clean_line.check_case_error.is_case_error):
(_EnumState.process_clean_line.check_member_list):
(_EnumState):
(check_enum_members):
(check_style):
(CppChecker):
(_EnumState.process_clean_line.is_case_error): Deleted.
(check_enum_casing): Deleted.
* Tools/Scripts/webkitpy/style/checkers/cpp_unittest.py:
(NoNonVirtualDestructorsTest):

Canonical link: https://commits.webkit.org/286492@main
